src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1349 @@ static ColliderCylinderInit D_80B37A5C = {
+                Player* player2 = GET_PLAYER(globalCtx);
+                s16 angleDiff2 = player2->actor.shape.rot.y - this->actor.yawTowardsPlayer;
+
+                this->actor.world.rot.y = this->actor.shape.rot.y;
+
+                if ((this->actor.xzDistToPlayer <= 80.0f) && !Actor_OtherIsTargeted(globalCtx, &this->actor) &&
+                    (((globalCtx->gameplayFrames % 4) == 0) || (ABS(angleDiff2) < 0x38E0))) {
+                    EnWf_SetupSlash(this);
+                } else {
+                    EnWf_SetupRunAtPlayer(this, globalCtx);
+                }
+            }
+        }
+
+        if ((animPrevFrame != (s32)this->skelAnime.curFrame) && (animFrameSpeedDiff <= 0) &&
+            ((animSpeed + animPrevFrame) > 0)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_WOLFOS_WALK);
+            Actor_SpawnFloorDustRing(globalCtx, &this->actor, &this->actor.world.pos, 20.0f, 3, 3.0f, 50, 50, 1);
+        }
+
+        if ((globalCtx->gameplayFrames & 95) == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_WOLFOS_CRY);
+        }
+    }
+}
+
+void EnWf_SetupDie(EnWf* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gWolfosRearingUpFallingOverAnim, -4.0f);
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+
+    if (this->actor.bgCheckFlags & 1) {
+        this->unk_300 = false;
+        this->actor.speedXZ = -6.0f;
+    } else {
+        this->unk_300 = true;
+    }
+
+    this->action = WOLFOS_ACTION_DIE;
+    this->actor.flags &= ~1;
+    this->actionTimer = this->skelAnime.animLength;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_WOLFOS_DEAD);
+    EnWf_SetupAction(this, EnWf_Die);
+}
+
+void EnWf_Die(EnWf* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 2) {
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 0.5f, 0.0f);
+        this->unk_300 = false;
+    }
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0xD0);
+
+        if (this->switchFlag != 0xFF) {
+            Flags_SetSwitch(globalCtx, this->switchFlag);
+        }
+
+        Actor_Kill(&this->actor);
+    } else {
+        s32 i;
+        Vec3f pos;
+        Vec3f velAndAccel = { 0.0f, 0.5f, 0.0f };
+
+        this->actionTimer--;
+
+        for (i = ((s32)this->skelAnime.animLength - this->actionTimer) >> 1; i >= 0; i--) {
+            pos.x = Rand_CenteredFloat(60.0f) + this->actor.world.pos.x;
+            pos.z = Rand_CenteredFloat(60.0f) + this->actor.world.pos.z;
+            pos.y = Rand_CenteredFloat(50.0f) + (this->actor.world.pos.y + 20.0f);
+            EffectSsDeadDb_Spawn(globalCtx, &pos, &velAndAccel, &velAndAccel, 100, 0, 255, 255, 255, 255, 0, 0, 255, 1,
+                                 9, true);
+        }
+    }
+}
+
+void EnWf_UpdateHeadRot(EnWf* this, GlobalContext* globalCtx) {
```

Comment:
since you unnamed `headRot` I guess this function should be unnamed too, as heartbreaking as it is